react-js/react-hooks.md

```diff
@@ -2,89 +2,284 @@
+ ```
+const HeadlineComponent = props => (<h1>{props.headline}</h1>)
+``` 
+
+#### Why were hooks introduced?
+Read [Intro to react hooks](https://reactjs.org/docs/hooks-intro.html) (5-10 minutes). 
+
+ Note:  React hooks were introduced in version 16.8. Previous alternatives to hooks include [render props](https://reactjs.org/docs/render-props.html),
+ [higher-order components](https://reactjs.org/docs/higher-order-components.html) such as the HOC used to connect a component to [the redux store](https://code.likeagirl.io/how-to-create-a-connected-hoc-with-react-and-redux-8631f7b1497). 
+ Right now you don't have to understand all of these alternatives in depth. The main point to note is there were a lot of solutions for the same problem.Hooks introduce *one* concept that solves these problems. 
+
+#### How do I use hooks?
+1. Read [Hooks Overview](https://reactjs.org/docs/hooks-overview.html)
+2. Walk through the examples on [Using the State Hook](https://reactjs.org/docs/hooks-state.html) to understand the state hook and its Class equivalent (in the counter `Example` component ).  Use [this codepen]([https://codepen.io/dpenny/pen/RwPPVqY](https://codepen.io/dpenny/pen/RwPPVqY)), to play around with these two ways of creating the same functionality. 
+
+*Discuss with a partner:*
+How do hooks achieve the same logic as the class component? Can you see how the code with hooks is cleaner?
 
-Build on the first information. Have apprentices guess things, do an activity, etc.
-
-Make sure to mention these things:
-
-- Things
-	- This is a sub-thing
-- More things
-- Even more things
-- Even more things
+### Guided Practice
+**Goal:  understand how we would use React hooks to implement a simple to-do list.**
+
+A user should be able to:
+-  Add items (via text input field)
+-  Press a button 'All done' which will clear all todo items, and a window alert will say 'Good job!'
+- Press a button 'Clear all' which will clear all todo items
+- When there are more than 5 items on the list, there should be a status message on the top that says 'Try to complete current tasks before adding new ones! '
+This will walk through one possible approach to solving this using React hooks. Note that there is no one right answer or one right approach! 
+
+Please fork [this template](https://codepen.io/dpenny/pen/rNVVZrb) and follow along with this tutorial.
+
+0. **What should the component look like?**
+We can start with a 'sketch' of what our component might look like. This does not need UI or styling. We start with an outline so that we can think about what information and functionality this component will need. 
+
+For example,
+``` 
+const {useState, useEffect} = React;
+
+const TodoApp = () => {
+  return (
+    <div>
+      <h3>status here</h3>
+      <div>
```

Comment:
Do you think it would be better to model a proper `<form>`? I think semantic markup is super important, and we should model best practices.